### PR TITLE
Bug fix for illegal character from video title.

### LINKED ISSUE #4: Error when the video title contains some illegal character
Some videos have illegal character like `/\?:`, etc, are causing errors. [Playlist like this](https://www.youtube.com/playlist?list=PLi8AeYwpQ7LiQ3t2eiijG09JnSnyHKGmx) is the example.

This causing error when trying to load the 3rd video.

Maybe add a Character Escaper? Thank you.

### PR DESCRIPTION
**Pull request by chez14**
This commit will fix #4.

File name taken from the video title, in which sometimes people put illegal character to the title.
This commit will remove it, and cleanse the remaining unneeded whitespaces.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -15,9 +15,15 @@ var downloadVideo = function(params, callback) {
...
-	}, Math.ceil(Math.random()*30));*/
+	/*
+	setTimeout(function() {
+	callback();
+	}, Math.ceil(Math.random()*30));
+	*/
+}
+
+var filterFileName = function(fileName){
+	return (fileName.replace(/[\\\/\:\*\?\"\<\>\|]+/, "").replace(/\s+/, " "));
```
Comment (RoobinGood) [inline app.js]: Why do you bracket all expression?

Comment (RoobinGood) [inline app.js]: You regular expressions in `replace` will change only first occurrence. Use 'g' option to replace all occurrences.

File: app.js [hunk 2]
```diff
@@ -15,9 +15,15 @@ var downloadVideo = function(params, callback) {
 		.on('finish', callback);
 
 	// test
-	/*setTimeout(function() {
-		callback();
-	}, Math.ceil(Math.random()*30));*/
+	/*
+	setTimeout(function() {
+	callback();
```
Comment (RoobinGood) [inline app.js]: What happened with indents? )

File: app.js [hunk 3]
```diff
@@ -15,9 +15,15 @@ var downloadVideo = function(params, callback) {
...
+	}, Math.ceil(Math.random()*30));
+	*/
+}
+
+var filterFileName = function(fileName){
```
Comment (RoobinGood) [inline app.js]: I think that we can give better name to this function cause it not only about filter illegal symbols. What about `normalizeFileName` or `fixFileName`?

File: app.js (hunk 1 above)
Comment (chez14) [reply app.js]: oh i'm sorry hahahaha i'm new on regexing things in javascript xD

File: app.js (hunk 3 above)
Comment (chez14) [reply app.js]: hmmm seems good, will take `normalizeFileName`.

**Comment by chez14** [comment]
Hmm, what do you think for this commit?

**Comment by RoobinGood** [comment]
Good, thank you!

**Pull request merged**